### PR TITLE
Adding mobile device crash parsing to Crashes table

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**
This commit adds mobile device crashes to the list of crash logs parsed
by the Crashes table as well as adding a lamdba to improve code reuse.

### CODE REVIEW

File: osquery/tables/system/darwin/crashes.cpp [hunk 1]
```diff
@@ -85,12 +93,18 @@ void readCrashDump(const std::string& app_log, Row& r) {
...
     // Process and grab all register values
-    if (toks[0] == "rax") {
-      r["registers"] = *it + *(++it);
+    if (toks[0] == "rax" || toks[0] == "x0") {
+      std::string reg_str = *it + " " + *(++it);
+
+      alg::replace_all(reg_str, ": ", ":");
+      alg::replace_all(reg_str, "   ", " ");
+
+      r["registers"] = reg_str;
```
Comment (theopolis) [inline osquery/tables/system/darwin/crashes.cpp]:
```
r["registers"] = std::move(reg_str);
```

**Comment by osqueryer** [comment]
Build origin/pr/2076/merge (Job: 2644) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2076/merge (Job: 2645) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2076/merge (Job: 2646) is good! :+1:

**Pull request merged**